### PR TITLE
[Merged by Bors] - feat(Topology/Inseparable): Define `specializing` maps between topological spaces

### PR DESCRIPTION
**Pull request by erdOne**
A port of https://github.com/leanprover-community/mathlib/pull/17113

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [inline Mathlib/Topology/Inseparable.lean:236]:
```suggestion
  fun _ _ h => h.map hf
```
and (almost) everywhere below
(I don't think term-mode proof/defs are exempt from indentation ...)

File: Mathlib/Order/UpperLower/Basic.lean
Comment (alreadydone) [inline Mathlib/Order/UpperLower/Basic.lean:1714]:
```suggestion
@[simp] lemma upperClosure_eq :
    ↑(upperClosure s) = s ↔ IsUpperSet s :=
  ⟨(· ▸ UpperSet.upper _), IsUpperSet.upperClosure⟩

@[simp] lemma lowerClosure_eq :
    ↑(lowerClosure s) = s ↔ IsLowerSet s :=
  @upperClosure_eq αᵒᵈ _ _
```

File: Mathlib/Order/UpperLower/Basic.lean
Comment (alreadydone) [inline Mathlib/Order/UpperLower/Basic.lean:456]:
```suggestion
  ⟨fun h x ↦ (isLowerSet_Iic x).image_fibration h, fun H x _ e ↦ H x e ⟨x, le_rfl, rfl⟩⟩
```
and everywhere below

**Comment by alreadydone** [comment]
Since I approved the mathlib3 PR, I'd approve this too once indentation is fixed.

File: Mathlib/Topology/Inseparable.lean
Comment (erdOne) [reply Mathlib/Topology/Inseparable.lean:236]: I thought tactics were indented because they were in a `by`-block, and terms are not in a block? 
I'll indent them anyway.

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [inline Mathlib/Topology/Inseparable.lean:369]: indent

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [inline Mathlib/Topology/Inseparable.lean:379]: indent

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [inline Mathlib/Topology/Inseparable.lean:395]: indent

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [inline Mathlib/Topology/Inseparable.lean:428]: indent

**Review by alreadydone — COMMENTED** [review]
You're close, just four are remaining!

File: Mathlib/Topology/Inseparable.lean
Comment (alreadydone) [reply Mathlib/Topology/Inseparable.lean:395]:
```suggestion
lemma IsClosedMap.specializingMap (hf : IsClosedMap f) : SpecializingMap f :=
  specializingMap_iff_stableUnderSpecialization_image_singleton.mpr <|
    fun _ ↦ (hf _ isClosed_closure).stableUnderSpecialization
```

**Review by alreadydone — APPROVED** [review]
Thanks!
maintainer merge

**Comment by semorrison** [comment]
bors merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
